src/useContext/IsAuthContext.js

```diff
@@ -0,0 +1,26 @@
+import React, { useState, useContext, createContext } from "react";
+
+const IsAuthContext = createContext();
+const IsAuthUpdateContext = createContext();
+
+export function useAuth() {
+  return useContext(IsAuthContext);
+}
+
+export function useAuthUpdate() {
+  return useContext(IsAuthUpdateContext);
+}
```

Comment:
You don't have to create a context for each value you plan on using, in your case `isAuth` and `setIsAuth`.

You can pass both of them in a single provider by passing them as an object.

```js
<IsAuthContext.Provider value={{
  isAuth,
  setIsAuth,
}}>
```

Reply:
Thanks a lot for the clarification Chris, I've fixed it